pkg/virt-api/validating-webhook/validating-webhook.go

```diff
@@ -321,6 +321,25 @@ func validateVirtualMachineSpec(field *k8sfield.Path, spec *v1.VirtualMachineSpe
 		return causes
 	}
 
+	// Validate memory size if values are not negative
+	if spec.Domain.Resources.Requests.Memory().Value() < 0 {
+		causes = append(causes, metav1.StatusCause{
+			Type: metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("%s '%s': must be greater than or equal to 0.", field.Child("domain", "resources", "requests", "memory").String(),
+				spec.Domain.Resources.Requests.Memory()),
+			Field: field.Child("domain", "resources", "requests", "memory").String(),
+		})
+	}
+
+	if spec.Domain.Resources.Limits.Memory().Value() < 0 {
+		causes = append(causes, metav1.StatusCause{
+			Type: metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("%s '%s': must be greater than or equal to 0.", field.Child("domain", "resources", "limits", "memory").String(),
+				spec.Domain.Resources.Limits.Memory()),
+			Field: field.Child("domain", "resources", "limits", "memory").String(),
+		})
+	}
+
```

Comment:
Do we want also to validate that limit is greater than the request?

Reply:
Good point. we need that check too.

Reply:
By the way, do we have a possibility to validate virt-launcher pod spec with k8s validator on VM validation stage?
It can save use some job.

Reply:
I have added validation if limit is greater than request